compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -320,6 +320,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             infer_args,
         );
 
+        // Skip if type isn't generic
+        if generic_params.params.len() == 0 {
+            return (tcx.intern_substs(&[]), vec![], arg_count);
```

Comment:
Is this not ignoring associated type bindings?

Reply:
The following is from the doc comment of this function:

```
    /// 3. The `generic_args` contains info on the `<...>` contents. The `usize` type
    ///    parameters are returned in the `SubstsRef`, the associated type bindings like	    ///    parameters are returned in the `SubstsRef`, the associated type bindings like
    ///    `Output = u32` are returned in the `Vec<ConvertedBinding...>` result.	    ///    `Output = u32` are returned in the `Vec<ConvertedBinding...>` result.
```

I would guess that if there are no generic params, then we aren't ignoring anything. My concern is that I don't know whether `generic_args` and `generic_params` refer to the same thing.

Reply:
Thanks - now it shouldn't :) I wonder why there weren't tests to cover that, though.

Reply:
Yes, I'm concerned there weren't any. Could you add a test for this, and make sure that it does fail with your previous code?

Reply:
I can try but I'm not entirely sure if I understand what needs to be tested.

Reply:
Presumably some situation where you have associated types, but no generic parameters. But that may not quite be enough, because I'm surprised there aren't already situations like this in the test suite...